From d1b3723c385f62b227829b87cebd2d88dd6b5b1b Mon Sep 17 00:00:00 2001
From: Chad Dupuis <cdupuis@redhat.com>
Date: Thu, 21 Aug 2014 12:58:35 -0400
Subject: [ethernet] qlge: remove unnecessary pci_set_drvdata()

Message-id: <1408625927-3701-5-git-send-email-cdupuis@redhat.com>
Patchwork-id: 88533
O-Subject: [PATCH RHEL 7.1 04/16] net: qlge: remove unnecessary pci_set_drvdata()
Bugzilla: 1089357
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jingoo Han <jg1.han@samsung.com>

Bug 1089357 (https://bugzilla.redhat.com/show_bug.cgi?id=1089357)

net commit id f9c4cd97b600c0e404e98cec7326e15c2e394480

The driver core clears the driver data to NULL after device_release
or on probe failure. Thus, it is not needed to manually clear the
device driver data to NULL.

Signed-off-by: Jingoo Han <jg1.han@samsung.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/qlogic/qlge/qlge_main.c b/drivers/net/ethernet/qlogic/qlge/qlge_main.c
index 09822d5..42dac01 100644
--- a/drivers/net/ethernet/qlogic/qlge/qlge_main.c
+++ b/drivers/net/ethernet/qlogic/qlge/qlge_main.c
@@ -4579,7 +4579,6 @@ static void ql_release_all(struct pci_dev *pdev)
   iounmap(qdev->doorbell_area);
  vfree(qdev->mpi_coredump);
  pci_release_regions(pdev);
- pci_set_drvdata(pdev, NULL);
 }
 
 static int ql_init_device(struct pci_dev *pdev, struct net_device *ndev,
-- 
1.7.1